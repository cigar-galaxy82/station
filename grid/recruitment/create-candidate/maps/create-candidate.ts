/* eslint-disable jest/no-export, jest/valid-describe, jest/valid-title, jest/no-identical-title */

import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';
import { readFileSync } from 'fs';

import { buildSuperfaceTest } from '../../../test-config';

const sampleCandidate = {
  name: 'John Doe',
  firstName: 'John',
  lastName: 'Doe',

  email: 'john.doe@fakemail.com',
  phone: '1-859-557-6573',
  address: '25772 Gustave Shore, Iowa, USA',

  education: [
    {
      degree: 'MBA',
      school: 'University of Pennsylvania',
      fieldOfStudy: null,
      startedAt: '2008-03-01',
      endedAt: '2011-03-30',
    },
    {
      degree: 'B.S.',
      school: 'University of Chicago',
      fieldOfStudy: 'Marketing Communication & Economics',
      startedAt: '2004-09-01',
      endedAt: '2007-03-30',
    },
  ],

  workExperience: [
    {
      position: 'Sales Director',
      summary: 'Summary of a work experience at Test Company',
      company: 'Test Company',
      industry: 'Telecommunications',
      current: false,
      startedAt: '2011-03-01',
      endedAt: '2014-03-30',
    },
  ],

  cv: {
    name: 'cv-sample.pdf',
    data: readFileSync('./grid/recruitment/cv-sample.pdf', {
      encoding: 'base64',
    }),
  },

  links: [
    {
      name: 'twitter',
      url: 'https://twitter.com/candidate.username',
    },
    {
      name: 'Portfolio',
      url: 'https://url.to.portfolio',
    },
  ],
};

const describeIf = (
  condition: boolean,
  ...args: Parameters<jest.Describe>
): void => (condition ? describe(...args) : describe.skip(...args));

export const createCandidateTest = (
  provider: string,
  jobIds: { valid: string; invalid: string },
  options?: RecordingProcessOptions
): void => {
  describe(`recruitment/create-candidate/${provider}`, () => {
    describe('CreateCandidate', () => {
      let superface: SuperfaceTest;

      beforeAll(() => {
        jest.setTimeout(10000);
        superface = buildSuperfaceTest({
          profile: 'recruitment/create-candidate',
          provider,
          useCase: 'CreateCandidate',
        });
      });

      describe('when specified job does exist', () => {
        it('performs correctly', async () => {
          const result = await superface.run(
            {
              input: {
                jobId: jobIds.valid,
                ...sampleCandidate,
              },
            },
            options
          );

          expect(result.isOk()).toBeTruthy();
          expect(result).toMatchSnapshot();
        });
      });

      describe('when specified job does not exist', () => {
        it('returns error', async () => {
          const result = await superface.run(
            {
              input: {
                jobId: jobIds.invalid,
                firstName: 'Demo',
                lastName: 'Testing',
                email: 'demo_testing@fakemail.com',
              },
            },
            options
          );

          expect(() => result.unwrap()).toThrow();
          expect(result).toMatchSnapshot();
        });
      });

      describeIf(
        provider === 'breezy-hr',
        'when specified company does not exist',
        () => {
          let companyId: string;

          beforeAll(() => {
            companyId = process.env.BREEZY_HR_COMPANY_ID!;

            process.env.BREEZY_HR_COMPANY_ID = '1b111c1111ef11';
          });

          afterAll(() => {
            process.env.BREEZY_HR_COMPANY_ID = companyId;
          });

          it('returns error', async () => {
            const result = await superface.run(
              {
                input: {
                  jobId: jobIds.valid,
                  firstName: 'Demo',
                  lastName: 'Testing',
                  email: 'demo_testing@fakemail.com',
                },
              },
              options
            );

            expect(() => result.unwrap()).toThrow();
            expect(result).toMatchSnapshot();
          });
        }
      );

      describeIf(
        provider === 'workable',
        'when specified subdomain does not exist',
        () => {
          let subdomain: string;

          beforeAll(() => {
            subdomain = process.env.WORKABLE_SUBDOMAIN!;

            process.env.WORKABLE_SUBDOMAIN = 'invalid-superface';
          });

          afterAll(() => {
            process.env.WORKABLE_SUBDOMAIN = subdomain;
          });

          it('returns error', async () => {
            const result = await superface.run(
              {
                input: {
                  jobId: jobIds.valid,
                  firstName: 'Demo',
                  lastName: 'Testing',
                  email: 'demo_testing@fakemail.com',
                },
              },
              options
            );

            expect(() => result.unwrap()).toThrow();
            expect(result).toMatchSnapshot();
          });
        }
      );
    });
  });
};
