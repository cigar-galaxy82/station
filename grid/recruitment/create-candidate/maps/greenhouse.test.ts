import { createCandidateTest } from './create-candidate';
import { createCandidateFeaturesTest } from './create-candidate-features';

const GREENHOUSE_PROVIDER_NAME = 'greenhouse';

createCandidateTest(GREENHOUSE_PROVIDER_NAME, {
  valid: '4031896006',
  invalid: '-1',
});

createCandidateFeaturesTest(GREENHOUSE_PROVIDER_NAME);
