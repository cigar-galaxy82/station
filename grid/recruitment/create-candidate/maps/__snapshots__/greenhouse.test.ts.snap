// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recruitment/create-candidate/greenhouse CreateCandidate when specified job does exist when valid inputs passed performs correctly 1`] = `
Ok {
  "value": Object {
    "id": "5884696006",
    "jobId": "4031896006",
  },
}
`;

exports[`recruitment/create-candidate/greenhouse CreateCandidate when specified job does not exist returns error 1`] = `
Err {
  "error": "MappedHTTPError: Expected HTTP error
Properties: {
  \\"title\\": \\"Unprocessable Entity\\",
  \\"detail\\": {
    \\"errors\\": [
      {
        \\"message\\": \\"A job with this id does not exist.\\",
        \\"field\\": \\"job_id\\"
      }
    ]
  },
  \\"code\\": \\"BadRequest\\"
}
AST Path: definitions[0].statements[1].responseHandlers[2].statements[0]
Original Map Location: Line 73, column 7",
}
`;

exports[`recruitment/create-candidate/greenhouse CreateCandidateFeatures returns valid greenhouse features 1`] = `
Ok {
  "value": Object {
    "cvMIMETypes": Array [
      "application/atom+xml",
      "application/javascript",
      "application/json",
      "application/msgpack",
      "application/msword",
      "application/pdf",
      "application/rss+xml",
      "application/vnd.ms-excel",
      "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
      "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
      "application/vnd.ms-powerpoint",
      "application/xml",
      "application/x-www-form-urlencoded",
      "application/x-yaml",
      "application/zip",
      "multipart/form-data",
      "image/bmp",
      "image/gif",
      "image/jpeg",
      "image/png",
      "image/tiff",
      "text/calendar",
      "text/css",
      "text/csv",
      "text/html",
      "text/javascript",
      "text/plain",
      "text/vcard",
      "video/mpeg",
    ],
    "cvUploadMethods": Array [
      "url",
      "file",
    ],
  },
}
`;
